### ISSUE TITLE
No error or warning on duplicate CompiledName

### ISSUE THREAD

**Issue by kerams** [fable-compiler/Fable#4737]
https://fable.io/repl/#?code=C4TwDgpgBAGlC8BYAUFNUA+UDaAeAwgPYC2YAlgDYQAmAcgIbHQBEAZq8wHwC6UAginSYcBEuSp1GLdl14AhIA&html=Q&css=Q

```fsharp
type X =
    | [<CompiledName "ff">] A
    | [<CompiledName "ff">] B
```

Fable, unlike .NET, honours this attribute on union cases. However, it doesn't verify that the names are unique. At the very least I would like to see a warning if there are valid use cases for this, otherwise an error.

**Comment by MangelMaxime** [comment]
I am almost certain that I saw some JavaScript libraries that used different value of enum to map to the same value.

In those scenarios it is normal case and does not need a warning or error message as they would be false positive.

Because of that I am unsure if this is something we can improve.

**Comment by kerams** [comment]
You can disable a warning, but you can't conjure one out of thin air:). I am starting to make heavy use of `CompiledName` and I want compile-time assurance that I didn't make a mistake in duplicating labels.

**Comment by MangelMaxime** [comment]
> You can disable a warning, but you can't conjure one out of thin air

Fable does not have this feature (yet)

Also, I confused `CompiledName` with `CompiledValue` which is something else.

And indeed, I think have 2 enum cases name being the same is probably strange and should be reported as error. If there are issues with that in the future, we can reduce it to a warning.

**Comment by MangelMaxime** [comment]
While working on the fix for this issue I found one Fable library using this feature:

[Fable.React.Props.fs](https://github.com/fable-compiler/fable-react/blob/b3dc782001c6db4d24c526ff401a99ab571f1d8d/src/Fable.React/Fable.React.Props.fs#L31-L37)

```fs
type Prop =
    | Key of string
    | Ref of (Element->unit)
    /// To be used in combination with `useRef` hook
    | [<System.Obsolete("Use RefValue")>] [<CompiledName("ref")>] RefHook of IRefHook<Element option>
    | [<CompiledName("ref")>] RefValue of IRefValue<Element option>
    interface IHTMLProp
```

I am still doing more investigation
